MAUI/Popup/working-with-mvvm.md

```diff
@@ -0,0 +1,444 @@
+
+{% tabs %}
+{% highlight xaml tabtitle="MainPage.xaml" hl_lines="13 15" %}
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             xmlns:sfPopup="clr-namespace:Syncfusion.Maui.Popup;assembly=Syncfusion.Maui.Popup"
+             xmlns:local="clr-namespace:PopupMVVM"
+             x:Class="PopupMVVM.MainPage">
+    <ContentPage.BindingContext>
+        <local:ViewModel/>
+    </ContentPage.BindingContext>
+    <ContentPage.Content>
+        <StackLayout Padding="20">
+            <Button x:Name="clickToShowPopup" Text="ClickToShowPopup" 
+                    VerticalOptions="Start" HorizontalOptions="Center" 
+                    Command="{Binding OpenPopupCommand}" />
+            <sfPopup:SfPopup x:Name="sfPopup"
+                             IsOpen="{Binding IsOpen}">
+            </sfPopup:SfPopup>
+        </StackLayout>
+    </ContentPage.Content>
+</ContentPage>
+{% endhighlight %}
+{% endtabs %}
+
+{% tabs %}
+{% highlight c# tabtitle="ViewModel.cs" %}
+public class ViewModel : INotifyPropertyChanged
+{
+    private bool isOpen;
+    public ICommand OpenPopupCommand { get; set; }
+
+    public bool IsOpen
+    {
+        get
+        {
+            return isOpen;
+        }
+        set
+        {
+            isOpen = value;
+            RaisePropertyChanged("IsOpen");
+        }
+    }
+
+    public ViewModel()
+    {
+        OpenPopupCommand = new Command(OnOpenPopup);
+    }
+
+    private void OnOpenPopup()
+    {
+        IsOpen = true;
+    }
+
+    #region INotifyPropertyChanged
+
+    public event PropertyChangedEventHandler PropertyChanged;
+
+    private void RaisePropertyChanged(string propertyName)
+    {
+        if (this.PropertyChanged != null)
+        {
+            this.PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
+        }
+    }
+
+    #endregion
+}
+{% endhighlight %}
+{% endtabs %}
+
+## How to use commands for popup buttons?
+
+`SfPopup` provides command support for the popup buttons,
+
+ * [SfPopup.AcceptCommand](https://help.syncfusion.com/maui/popup/popup-events#accept-command)
+
+ * [SfPopup.DeclineCommand](https://help.syncfusion.com/maui/popup/popup-events#decline-command)
+
+## How to use MAUI popup as a page?
+
+To use `SfPopup` as a page, create a new XAML file with `SfPopup` in it. You can use this XAML file instance (SfPopup view), across any file in your project wherever you may want to display a popup, thus allowing you to reuse the same instance of the popup where you can modify or change the required properties alone based on the current context to be displayed.
+
+The below code snippet shows how to create a popup in XAML file. We have created a simple popup to show error message popup to user on a unsuccessful login.

```

Comment:
The following code sample shows how to create a popup in a XAML file. We have created a simple popup to display an error message popup to users on an unsuccessful login.